### DISCUSSION TITLE
new window while fullscreen video is playing

### DISCUSSION THREAD

**Discussion by veksha** [qtile/qtile#3729] [Q&A]
hi, all.
i have brave browser and youtube playing video in fullscreen (browser fullscreen function, not qtile fullscreen).
i press a key combination to spawn terminal and it is spawned and focused, but fullscreen video is still playing on very top hiding my new window behind.

can this behaviour be changed somehow? thanks.

**Comment by m-col** [comment]
If it's enough to just raise new windows to the top of the Z stack you can use a hook like this:
```py
from libqtile import hook

@hook.subscribe.client_managed
def _(window):
    window.cmd_bring_to_front()
```

**Reply by veksha** [reply]
thanks for suggestion.
this made all new windows appear in floating mode. (not good)
and new window is still not visible but focused. i can make it visible by starting to move/resize the window.

**Reply by veksha** [reply]
this kind of works. sending Esc key:

```python
@hook.subscribe.client_new
def _(window):
    #lazy.spawn("xdotool key Escape")
    subprocess.Popen(["xdotool", "key", "Escape"])
```

though I don't know why `lazy.spawn` is not working. only `subprocess.Popen`
do you know, @m-col ?

**Reply by veksha** [reply]
better code:
```python
@hook.subscribe.client_new
def _(window):
    wnd = subprocess.run(["xdotool", "search", "--name", "--onlyvisible", "brave"], capture_output=True, text=True).stdout
    if wnd:
        subprocess.run(["xdotool", "key", "Escape", "--window", wnd])
```

**Reply by m-col** [reply]
The lazy objects only represent commands for later use e.g. keybindings. If you just want the window ID which I guess is what xdotool is giving you, you can get that at `window.wid`

**Reply by veksha** [reply]
impoved code more. this will ensure that window is Fullscreen before sending ESC key:

```python
@hook.subscribe.client_new
def _(window):   
    wnd = subprocess.run(["sh", "-c", "xdotool getmouselocation | cut -d: -f5"], capture_output=True, text=True).stdout.strip()
    isfullscreen = subprocess.run(["sh", "-c", f"xprop -id {wnd} | grep FULLSCREEN"], capture_output=True, text=True).stdout.strip()
    if isfullscreen:
        subprocess.run(["xdotool", "key", "Escape", "--window", wnd])
```
